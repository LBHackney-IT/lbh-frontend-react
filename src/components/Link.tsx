import React from "react";
import classNames from "classnames";
import PropTypes from "prop-types";

import { ComponentRegister, LinkComponent } from "../ComponentRegister";

import "lbh-frontend/lbh/core/_links.scss";

/**
 * A component for creating an anchor linking to other pages. You can customize the component used internally for
 * integration with routers using {@link ComponentRegister.init}. By default, it uses an <a> tag.
 *
 * @param {string} [className]
 * @param {string} href
 * @param {ReactNode} children
 *
 * @returns {LinkComponent}
 */
export const Link: LinkComponent = ({ className, href, children }) => {
  return (
    <ComponentRegister.Link
      className={classNames("govuk-link lbh-link", className)}
      href={href}
    >
      {children}
    </ComponentRegister.Link>
  );
};

Link.propTypes = {
  className: PropTypes.string,
  href: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired
};
