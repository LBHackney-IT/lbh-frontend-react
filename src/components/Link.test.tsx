import React from "react";
import { create } from "react-test-renderer";

import { Link } from "./Link";

it("renders correctly with all props", () => {
  const component = create(
    <Link id="testId" className="class1" href="/">
      Link
    </Link>
  );
  expect(component).toMatchSnapshot();
});

it("renders with only required props", () => {
  const component = create(<Link href="/">Link</Link>);
  expect(component).toMatchSnapshot();
});
