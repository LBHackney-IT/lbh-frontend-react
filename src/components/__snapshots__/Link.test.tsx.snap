// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<a
  className="govuk-link lbh-link class1"
  href="/"
>
  Link
</a>
`;

exports[`renders with only required props 1`] = `
<a
  className="govuk-link lbh-link"
  href="/"
>
  Link
</a>
`;
